api/deploy.go

```diff
@@ -133,28 +131,28 @@ func (d *serviceDeployer) preprocessArchive(r io.Reader, path string) error {
 	return archive.Untar(r, path, nil)
 }
 
-func (d *serviceDeployer) processPath(path string) (io.ReadCloser, error) {
+func (d *serviceDeployer) processPath(path string) (string, error) {
 	// XXX: remove .git folder from repo.
 	// It makes docker build iamge id same between repo clones.
 	if err := os.RemoveAll(filepath.Join(path, ".git")); err != nil {
-		return nil, err
+		return "", err
 	}
 
 	// NOTE: this is check for tar repos, if there is only one
 	// directory inside untar archive set temp path to it.
 	dirs, err := ioutil.ReadDir(path)
 	if err != nil {
-		return nil, err
+		return "", err
 	}
 	if len(dirs) == 1 && dirs[0].IsDir() {
 		path = filepath.Join(path, dirs[0].Name())
 	}
 
-	return archive.Tar(path, archive.Uncompressed)
+	return path, nil
 }
 
 // deploy deploys a service in path.
-func (d *serviceDeployer) deploy(r io.Reader) (*service.Service, *importer.ValidationError, error) {
+func (d *serviceDeployer) deploy(path string) (*service.Service, *importer.ValidationError, error) {
```

Comment:
[NOT BLOCKING] This modif should have been done in another PR

Reply:
in that case we couldn't have remove `tempPath` private field from `Service` type which is the actual purpose of this PR. we needed to make `Service` type logicless by removing all private fields